pkg/controller/atlas/logger.go

```diff
@@ -0,0 +1,101 @@
+package atlas
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"os"
+	"strings"
+)
+
+func dumpRequest(req *http.Request) string {
+	if req == nil {
+		return "<nil>"
+	}
+	requestDump, err := httputil.DumpRequest(req, true)
+	if err != nil {
+		return err.Error()
+	}
+	return string(requestDump)
+}
+
+func dumpResponse(rsp *http.Response) string {
+	if rsp == nil {
+		return "<nil>"
+	}
+	responseDump, err := httputil.DumpResponse(rsp, true)
+	if err != nil {
+		return err.Error()
+	}
+	return string(responseDump)
+}
+
+// AtlasLog is all is needed to implement logging
+type AtlasLog interface {
```

Comment:
As already mentioned and adding what Igor said, you could configure the client to use zap logger to do this job.
I see no need to maintain our custom logger even for test purposes.

Reply:
Will fix.